pkg/roll/run_update.go

```diff
@@ -252,14 +252,14 @@ func (u *update) rollLoop(podID types.PodID, hChecks <-chan map[types.NodeName]h
 					"nextAdd":    nextAdd,
 				}).Infof("Adding %d new nodes and removing %d old nodes", nextAdd, nextRemove)
 				if nextRemove > 0 {
-					err = u.rcs.AddDesiredReplicas(u.OldRC, -nextRemove)
+					err = u.rcs.CASDesiredReplicas(u.OldRC, oldNodes.Desired, oldNodes.Desired-nextRemove)
 					if err != nil {
 						u.logger.WithError(err).Errorln("Could not decrement old replica count")
 						break
```

Comment:
It's fragile to abort the whole update if one operation has an error. What I'd like is for it to re-run its computation using the new RC state. With `break`, it will do that. Eventually. After the health update watch hits its wait time and re-sends all its data. That isn't ideal behavior, but I'll take it.